packages/react-dom/src/client/ToStringValue.js

```diff
@@ -44,14 +45,39 @@ export opaque type TrustedValue: {toString(): string, valueOf(): string} = {
   valueOf(): string,
 };
 
+export function evaluateToStringOrTrustedType(
```

Comment:
Does this need to be its own function? Can't it just be the same as `toStringOrTrustedType`?

`toStringOrTrustedType` is called in two other places.

This one seems like you might have missed: https://github.com/facebook/react/pull/17322/files#diff-214d5116dd8b5a7d184cdf5b2160f94cR183

Seems like it should also trigger this path.

The other one is here but I think that's actually already the wrong place to do this and it shouldn't toString this at all since the initial rendering doesn't: https://github.com/facebook/react/blob/9fe1031244903e442de179821f1d383a9f2a59f2/packages/react-dom/src/client/ReactDOMComponent.js#L792
But it doesn't hurt that we check for `REACT_OPAQUE_OBJECT_TYPE`. Even though the existing call is a bug in the first place.

That way we need less code overall.